mlrun/datastore/utils.py

```diff
@@ -73,11 +75,53 @@ def parse_kafka_url(
 
 
 def upload_tarball(source_dir, target, secrets=None):
-
     # will delete the temp file
     with tempfile.NamedTemporaryFile(suffix=".tar.gz") as temp_fh:
         with tarfile.open(mode="w:gz", fileobj=temp_fh) as tar:
             tar.add(source_dir, arcname="")
         stores = mlrun.datastore.store_manager.set(secrets)
         datastore, subpath = stores.get_or_create_store(target)
         datastore.upload(subpath, temp_fh.name)
+
+
+def filter_df_start_end_time(
+    df, time_column=None, start_time=None, end_time=None
```

Comment:
A type hint on `df` at least would be useful here, since you're doing `isinstance` on it, and it would help to figure out what it can be other than `pd.DataFrame` (reading further, it's supposed to be a `df` generator, so is there a specific type for it? Is it `Iterator[DataFrame]`?)